Readme.md

```
@@ -194,25 +231,49 @@ For production deployments in a VM:
+```fish
+make build
+make test
+```
+
+Major-version upgrades (for example `sigs.k8s.io/controller-runtime` v0.22 -> v0.24, or any `/v2`, `/v3` import path bump) often contain breaking API changes and should be reviewed one module at a time rather than via a blanket `make deps-update`.
+
+Docker base images in the per-component `Dockerfile`s are tracked by Renovate; for a manual bump, look up the latest tag on the relevant registry and edit the `FROM` line.
+
+## High-Level Architecture Diagram
 
 ```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)